File: 1277-count-square-submatrices-with-all-ones/1277-count-square-submatrices-with-all-ones.cpp
class Solution {
public:
    int countSquares(vector<vector<int>>& matrix) {
        int ans=0;
        
        for(int i=0; i<matrix.size(); i++){
            for(int j=0; j<matrix[i].size(); j++){
                if(i==0 || j==0){
                    ans+=matrix[i][j];
                } else {
                    int m=min(matrix[i-1][j],min(matrix[i][j-1],matrix[i-1][j-1]));
                    if(matrix[i][j]==1){
                        ans+=m+1;
                        matrix[i][j]=m+1;
                    }
                }
            }
        }
        
        return ans;
    }
};

File: 1277-count-square-submatrices-with-all-ones/1277-count-square-submatrices-with-all-ones.java
class Solution {
    public int countSquares(int[][] matrix) {
        int n=matrix.length;
        int m=matrix[0].length;
        int ans=0;
        for(int i=0; i<n; i++){
            for(int j=0; j<m; j++){
                if(matrix[i][j]==1){
                    if(i==0 || j==0){
                        ans+=1;
                    } else {
                        int size=Math.min(matrix[i-1][j],Math.min(matrix[i][j-1],matrix[i-1][j-1]));
                        ans+=size+1;
                        matrix[i][j]=size+1;
                    }
                }
            }
        }
        
        return ans;
    }
}